### PR TITLE
Testing missing branches

### PR DESCRIPTION
**Pull request by erickpintor**
I'm looking at the coverage report and fixing a few missing branches. This is still a WIP. I'll let you know once it's ready for review.

### CODE REVIEW

**Comment by erickpintor** [comment]
I think I've covered the branches I found most important to cover.
This PR should be ready to review.

**Comment by erickpintor** [comment]
Closes: https://github.com/faunadb/sales-engineering/issues/400

File: faunadb/field_test.go [hunk 1]
```diff
@@ -69,3 +85,16 @@ func assertFailToExtractField(t *testing.T, value Value, field Field, message st
 	var res Value
 	require.EqualError(t, value.At(field).Get(&res), message)
 }
+
+func assertFailToExtractAnyFieldFor(t *testing.T, values ...Value) {
```
Comment (marrony) [inline faunadb/field_test.go]: Is this function supposed to be reused? Seems to be used only once in `TestFailToExtractFieldForNonTransversableValues`

Comment (erickpintor) [inline faunadb/field_test.go]: It's not being reused. I end up extracting the function just so I could use the varargs to clean up the test a little.

File: faunadb/decode.go [hunk 2]
```diff
@@ -44,10 +44,6 @@ func newValueDecoder(source interface{}) *valueDecoder {
 func (c *valueDecoder) assign(value interface{}) error {
 	source, sourceType := indirectValue(value)
 
-	if !c.target.CanSet() {
```
Comment (marrony) [inline faunadb/decode.go]: What happens in this case now? It will panic the application?

**Comment by marrony** [comment]
lgtm

**Pull request merged**